meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -169,6 +179,87 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
         }
     }
 
+    /** 
+     * Find which the module the entity belongs to 
+
+     * @param entity
+     * @return MeveoModule, or null otherwise
+     */
+    @SuppressWarnings({ "rawtypes", "deprecation" })
+	public MeveoModule findModuleOf(P entity){
+    	MeveoModule module = null;
+    	if (entity != null) {
+    		Session session = this.getEntityManager().unwrap(Session.class);
+    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.ItemType = :itemType");
+    		q.setString("code", entity.getCode());
+    		q.setString("itemType", entity.getClass().getName());
+    		module = (MeveoModule) q.getResultList().get(0);
```

Comment:
Check if the list is not empty (return null if it's the case), or you will get index out of bounds exception